htlcswitch/link.go

```diff
@@ -2559,10 +2585,28 @@ func (l *channelLink) drivePendingStfuSends() {
 				handleErr(err)
 				return
 			}
+			l.tryResolveQuiescenceRequests()
 		})
 	}
 }
 
+// tryResolveQuiescenceRequests checks whether the quiescence protocol has
+// successfully completed. If it has, it completes the outstanding asynchronous
+// request, if there is one.
+func (l *channelLink) tryResolveQuiescenceRequests() {
+	if l.quiescer.isQuiescent() {
+		if resp := l.activeQuiescenceRequest; resp != nil {
+			ourTurn := l.quiescer.isLocallyInitiatedFinal()
```

Comment:
Now seeing this method called, perhaps we should rename it? To something like `stfuFinalized`? Which indicates that we're done now, so can send a message back to the caller.

Reply:
I will look into renaming it to something better.